products/jbrowse-web/src/Loader.tsx

```diff
@@ -570,12 +589,44 @@ const Renderer = observer(
                   }`,
                 )
               }
+            } else if (sessionSpec) {
+              try {
+                const { views } = sessionSpec
+                rootModel.setSession({
+                  name: `New session ${new Date().toLocaleString()}`,
+                })
+                const { session } = rootModel
+                Promise.all(
+                  views.map(async (view: any) => {
+                    const { tracks, type, assembly, loc } = view
+                    await rootModel.assemblyManager.waitForAssembly(assembly)
+                    if (type === 'LGV' || type === 'LinearGenomeView') {
+                      const materialView = session?.addView(
+                        'LinearGenomeView',
+                        {},
+                      )
+                      materialView.setWidth(800)
+                      materialView.navToLocString(loc, assembly)
+                      tracks.forEach((track: string) => {
+                        materialView.showTrack(track)
+                      })
+                    }
+                  }),
+                ).catch(e => {
+                  console.error(e)
+                  setError(e)
+                })
+              } catch (e) {
+                console.error(e)
+                rootModel.setDefaultSession()
+                rootModel.session?.notify(
+                  'Failed to load session spec from URL bar',
+                )
+              }
```

Comment:
Could we move this out of the loader into a separate file where all the session spec stuff lives

Reply:
Moved the whole SessionLoader model to a new file plus this helper function